interview-questions.md

```diff
@@ -8,50 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is a inputed value, index, or number in a function
+such as (array[bean, fruit, meat], /string"Hello How are you", index[2], 25) 
+where a argument an argument is if a value is given under a condition and then 
+it is logged in result as a return
+
+Researched answer: Parameters are the names listed in the functions Definition
+                    Example : function example(paramenter) {
+                        console.log(parameter)
+                    }
+                    An Argument are real values passed to the function
+                    const argument = "LEARN"
 
-Researched answer:
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: .map goes through each value in a array from a function and does not change the original array key and values are the predefined parameters
 
-Researched answer:
+Researched answer: .map is a collections of key value pairs. It does not contain any keys by default. it only contains what is explicitly put into it. It can be any value like functions objecs or any primitive. It iterates in the order of insertion.
```

Comment:
See what you can find about .map parameters. Key value pairs are found in objects. .map is a HOF